src/main/java/mira/Task.java

```diff
@@ -0,0 +1,79 @@
+package mira;
+
+/**
+ * Represents a task with a description and completion status.
+ */
+public abstract class Task {
+    private final String description; // The task's description
+    private boolean isDone; // Indicates if the task is completed
+
+    /**
+     * Constructs a new Task with the specified description.
+     * The task is initially marked as not done.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructs a new Task with the specified description and isDone.
+     *
+     * @param description The description of the task.
+     * @param isDone The status of the task.
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Sets the completion status of the task.
+     *
+     * @param status True to mark the task as done, false to mark it as not done.
+     */
+    public void setStatus(boolean status) {
+        this.isDone = status;
+    }
+
+    /**
+     * Returns the status icon representing whether the task is done.
+     * An "X" indicates the task is done; a space indicates it is not done.
+     *
+     * @return A string containing "X" if the task is done, otherwise a space.
+     */
+    private String getStatusIcon() {
+        return (this.isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Return the string format of task to be saved for storage.
```

Comment:
Maybe "Return" should be "Returns"? Just to keep consistency with the other documentations.